components/core/src/clp/GrepCore.hpp

```diff
@@ -0,0 +1,424 @@
+#ifndef CLP_GREPCORE_HPP
+#define CLP_GREPCORE_HPP
+
+#include <cstddef>
+#include <cstdint>
+#include <optional>
+#include <string>
+#include <string_view>
+#include <unordered_set>
+#include <vector>
+
+#include <log_surgeon/Lexer.hpp>
+#include <string_utils/string_utils.hpp>
+
+#include "Defs.h"
+#include "EncodedVariableInterpreter.hpp"
+#include "ir/parsing.hpp"
+#include "ir/types.hpp"
+#include "Query.hpp"
+#include "QueryToken.hpp"
+
+namespace clp {
+class GrepCore {
+public:
+    // Methods
+    /**
+     * Processes a raw user query into a Query.
+     *
+     * Note: callers are responsible for ensuring that the search string does not contain repeated
+     * wildcards "**" e.g. by using `clp::string_utils::clean_up_wildcard_search_string`.
```

Comment:
```suggestion
     * NOTE: Callers are responsible for ensuring that the search string is "clean", where clean is
     * defined by the criteria in `clean_up_wildcard_search_string`.
```

It's usually clearer and more consistent to use `NOTE` rather than `Note` (similar to `TODO`).